lib/lrmd/lrmd_client.c

```diff
@@ -769,7 +769,7 @@ lrmd_api_is_connected(lrmd_t * lrmd)
  * \param[in]  op            Name of API command to send
  * \param[in]  data          Command data XML to add to the sent command
  * \param[out] output_data   If expecting a reply, it will be stored here
- * \param[in]  timeout       Timeout in milliseconds (if 0, defaults to 1000);
+ * \param[in]  timeout       Timeout in milliseconds (if 0, defaults to 5000);
```

Comment:
Unfortunately it looks more complicated than this. This function uses the argument twice:

* It is passed to lrmd_create_op(), which adds it to the API command XML as F_LRMD_TIMEOUT, which the executor will parse into ((lrmd_cmd_t*)cmd)->timeout. I don't see anywhere that the executor defaults 0 there to some other value, so it may be left alone, which would likely affect log messages and systemd action retries, and potentially get passed to the stonith API's monitor (which would indeed end up defaulting to 1000) or resources_action_create(), which does who knows what with it.

* It is passed to lrmd_send_xml(), which can pass it to either crm_ipc_send() (which defaults 0 to 5000) or lrmd_tls_send_recv() (which defaults 0 to 10000)

Easiest solution here would be to just leave off the parenthesized bit, or, with a bit of faith, "(if 0, defaults to a reasonable value)".

Reply:
True, I'd cast an opposite bias (just as bad), thanks for catching that.